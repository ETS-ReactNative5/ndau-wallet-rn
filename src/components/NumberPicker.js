import React, { Component } from 'react'

import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import cssStyles from '../css/styles'
import FontAwesome5Pro from 'react-native-vector-icons/FontAwesome5Pro'
import styleConstants from '../css/styleConstants'

class UnlockModalDialog extends Component {
  render () {
    return (
      <View style={styles.container}>
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'center'
          }}
        >
          <TouchableOpacity onPress={this.props.subtractNumber}>
            <FontAwesome5Pro
              name='minus-circle'
              color={styleConstants.ICON_GRAY}
              size={32}
              light
            />
          </TouchableOpacity>
          <Text style={styles.text}>{this.props.number}</Text>
          <TouchableOpacity onPress={this.props.addNumber}>
            <FontAwesome5Pro
              name='plus-circle'
              color={styleConstants.ICON_GRAY}
              size={32}
              light
            />
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

var styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignContent: 'center'
  },
  text: {
    marginLeft: wp('10%'),
    marginRight: wp('10%'),
    color: '#ffffff',
    fontSize: 32,
    fontFamily: 'TitilliumWeb-Regular'
  }
})

export default UnlockModalDialog
