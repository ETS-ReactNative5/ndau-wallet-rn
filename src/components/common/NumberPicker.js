import React, { Component } from 'react'

import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import Icon from 'react-native-fontawesome-pro'
import AppConstants from '../../AppConstants'

class NumberPicker extends Component {
  render () {
    return (
      <View style={styles.container}>
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'center'
          }}
        >
          <Icon
            name='minus-circle'
            color={AppConstants.NUMBER_PICKER_COLOR}
            size={32}
            onPress={this.props.subtractNumber}
            type='light'
          />
          <Text style={styles.text}>{this.props.number}</Text>
          <Icon
            name='plus-circle'
            color={AppConstants.NUMBER_PICKER_COLOR}
            size={32}
            onPress={this.props.addNumber}
            type='light'
          />
        </View>
      </View>
    )
  }
}

var styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignContent: 'center'
  },
  text: {
    marginLeft: wp('10%'),
    marginRight: wp('10%'),
    color: '#ffffff',
    fontSize: 32,
    fontFamily: 'TitilliumWeb-Regular'
  }
})

export default NumberPicker
