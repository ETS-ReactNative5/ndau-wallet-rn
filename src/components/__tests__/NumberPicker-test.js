import { mount } from 'enzyme';
import React from 'react';
import { StyleSheet } from 'react-native';
import NumberPicker from '../NumberPicker';

import renderer from 'react-test-renderer';

describe('testing NumberPicker...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<NumberPicker parentStyles={styles} navigator={navigator} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
