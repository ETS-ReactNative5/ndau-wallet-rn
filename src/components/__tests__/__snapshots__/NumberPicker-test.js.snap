// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing NumberPicker... renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#A9A9A9",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 32,
          "marginLeft": 75,
          "marginRight": 75,
        }
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#A9A9A9",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
